test/c/test_circuit.c

```diff
@@ -0,0 +1,99 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+/**
+ * Test the zero constructor.
+ */
+int test_empty() {
+    QkCircuit *qc = qk_circuit_new(0, 0);
+    uint32_t num_qubits = qk_circuit_num_qubits(qc);
+    uint32_t num_clbits = qk_circuit_num_clbits(qc);
+    qk_circuit_free(qc);
+
+    return (num_qubits != 0 || num_clbits != 0) ? EqualityError : Ok;
+}
+
+int test_no_gate_1000_bits() {
+    QkCircuit *qc = qk_circuit_new(1000, 1000);
+    uint32_t num_qubits = qk_circuit_num_qubits(qc);
+    uint32_t num_clbits = qk_circuit_num_clbits(qc);
+    qk_circuit_free(qc);
+
+    return (num_qubits != 1000 || num_clbits != 1000) ? EqualityError : Ok;
+}
+
+int test_get_gate_counts_bv_no_measure() {
+    QkCircuit *qc = qk_circuit_new(1000, 1000);
+    double *params = NULL;
+    uint32_t i = 0;
+    uint32_t qubits[1] = {999};
+    qk_circuit_append_standard_gate(qc, QkStandardGate_XGate, qubits, params);
+    for (i = 0; i < 1000; i++) {
+        uint32_t qubits[1] = {i};
+        qk_circuit_append_standard_gate(qc, QkStandardGate_HGate, qubits, params);
+    }
+    for (i = 0; i < 1000; i += 2) {
+        uint32_t qubits[2] = {i, 999};
+        qk_circuit_append_standard_gate(qc, QkStandardGate_CXGate, qubits, params);
+    }
+    for (i = 0; i < 999; i++) {
+        uint32_t qubits[1] = {i};
+        qk_circuit_append_standard_gate(qc, QkStandardGate_HGate, qubits, params);
+    }
+    QkOpCounts op_counts = qk_circuit_count_ops(qc);
+    if (op_counts.len != 3) {
```

Comment:
op_counts and qc are not freed at failure 🙂

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/14006/commits/3e62ec6d8d4b68bb1521de0660659c214b7afa83 so it always cleans up after itself now.

Reply:
> The test doesn't fail :D

famous last words 😂 it's only been a few weeks writing C and I already have trust issues